crates/e2e-tests/src/fixtures/transactions.rs

```diff
@@ -0,0 +1,40 @@
+use alloy_consensus::{SignableTransaction, TxLegacy};
+use alloy_network::eip2718::Encodable2718;
+use alloy_primitives::{Address, Bytes, U256};
+use alloy_signer::Signer;
+use alloy_signer_local::PrivateKeySigner;
+use anyhow::Result;
+
+pub fn create_test_signer() -> PrivateKeySigner {
+    "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+        .parse()
+        .expect("Valid test private key")
+}
+
+pub async fn create_signed_transaction(
+    signer: &PrivateKeySigner,
+    to: Address,
+    value: U256,
+    nonce: u64,
+    gas_limit: u64,
+    gas_price: u128,
+) -> Result<Bytes> {
+    let tx = TxLegacy {
```

Comment:
just curious, any reason for `TxLegacy` over `TxEip1559`?

Reply:
sounds good! i think we can leave txlegacy for now and then if we actually need txeip1559 later we can just swap it out in a follow up